cognite/client/data_classes/_base.py

```diff
@@ -260,10 +260,21 @@ def _set_null(self, name):
         self._update_object[name] = {"setNull": True}
 
     def _add(self, name, value):
-        self._update_object[name] = {"add": value}
+        update_obj = self._update_object.get(name, {})
+        if "set" in update_obj:
+            del update_obj["set"]
+        if isinstance(value, list):
+            update_obj["add"] = update_obj.get("add", []) + value
+        else:
+            update_obj["add"] = {**update_obj.get("add", {}), **value}
+        self._update_object[name] = update_obj
 
     def _remove(self, name, value):
-        self._update_object[name] = {"remove": value}
+        update_obj = self._update_object.get(name, {})
+        if "set" in update_obj:
+            del update_obj["set"]
```

Comment:
I think this behaviour is very implicit and not necessarily very intuitive. I would rather this raised an error saying that you cannot remove from something you have already set.

Reply:
good idea!